Fix github releases action

krowvin opened the pull request:
This is an attempt to get the error here fixed so that we can auto create releases again. 
https://github.com/USACE-WaterManagement/repgen5/actions/runs/11280353941/job/31373141983


The package creation script for pypi also requires a release be made to run.

MikeNeilson commented:
You might consider updating this to what CDA uses:
https://github.com/USACE/cwms-data-api/blob/198140d1883731406c3eec5c8c45fff13ff3ad71/.github/workflows/build.yml#L64-L67


But perhaps also alter the action .yml to only do the publish on a tag and then you can just use the current tag.

krowvin commented:
Alright, how's that? Not sure about switching to 

`uses: softprops/action-gh-release@v2.0.8`
from
`uses: actions/create-release@v1.1.4`

Mostly just chasing the error that already existed to get it running again and making the release

MikeNeilson commented:
actions/create-release@v1.1.4 is no longer maintained, you may be having problems simply due to that. However if you want to try that first we can.

krowvin commented:
> is no longer maintained

Did not know that! Switched and also removed the tag creation - seems we don't need that if this only fires when a tag is made?
